doc/design/diagrams/iceoryx_software_layers_v3_0_0.puml

```diff
@@ -0,0 +1,38 @@
+@startuml
+
+title Eclipse iceoryx software layers (v3.0.0)
+
+skinparam BackgroundColor transparent
+skinparam componentStyle uml2
+
+skinparam component {
+   backgroundColor<<namespace>> yellow
+}
+
+skinparam package {
+   backgroundColor green
+}
+
+skinparam frame {
+   backgroundColor chocolate
+}
+
+[iceoryx_dust]
+[iceoryx_hoofs]
```

Comment:
You forgot the `iceoryx_platform` as lowest level. In the next days it will be moved out from hoofs and this building block ensures platform independent code and unifies the behavior of posix functionality across platforms.

Reply:
Currently, the `platform` library is located inside of iceoryx_hoofs and is represented here as module `Platform abstraction`. So it wouldn't make sense to list it now along with hoofs and posh.
I would suggest to modify the architecture accordingly when the migration actually happens in the Pull-Request.

Reply:
@elfenpiff Added platform library, ready for review.